docs/conf.py

```diff
@@ -0,0 +1,275 @@
+# pylint: disable=invalid-name
+# Requires Python 3.6+
+# Ref: https://www.sphinx-doc.org/en/master/usage/configuration.html
+"""Configuration for the Sphinx documentation generator."""
+
+import sys
+from functools import partial
+from pathlib import Path
+
+from setuptools_scm import get_version
+
+
+# -- Path setup --------------------------------------------------------------
+
+PROJECT_ROOT_DIR = Path(__file__).parents[1].resolve()  # pylint: disable=no-member
+get_scm_version = partial(get_version, root=PROJECT_ROOT_DIR)
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+
+
+sys.path.insert(0, str(PROJECT_ROOT_DIR))
+
+# Make in-tree extension importable in non-tox setups/envs, like RTD.
+# Refs:
+# https://github.com/readthedocs/readthedocs.org/issues/6311
+# https://github.com/readthedocs/readthedocs.org/issues/7182
+sys.path.insert(0, str((Path(__file__).parent / '_ext').resolve()))
+
+# -- Project information -----------------------------------------------------
+
+github_url = 'https://github.com'
+github_repo_org = 'abhinavsingh'
+github_repo_name = 'proxy.py'
+github_repo_slug = f'{github_repo_org}/{github_repo_name}'
+github_repo_url = f'{github_url}/{github_repo_slug}'
+github_sponsors_url = f'{github_url}/sponsors'
+
+project = github_repo_name.title()
+author = f'{project} project contributors'
+copyright = author  # pylint: disable=redefined-builtin
+
+# The short X.Y version
+version = '.'.join(
+    get_scm_version(
+        local_scheme='no-local-version',
+    ).split('.')[:3],
+)
+
+# The full version, including alpha/beta/rc tags
+release = get_scm_version()
+
+rst_epilog = f"""
+.. |project| replace:: {project}
+"""
+
+
+# -- General configuration ---------------------------------------------------
+
+
+# There are two options for replacing |today|: either, you set today to some
+# non-false value, then it is used:
+# today = ''
+# Else, today_fmt is used as the format for a strftime call.
+today_fmt = '%B %d, %Y'
+
+# The reST default role (used for this markup: `text`) to use for all
+# documents.
+# Ref: python-attrs/attrs#571
+default_role = 'any'
```

Comment:
@abhinavsingh this instructs Sphinx to link any text in single backticks to any role that has an object with that name. Practically, it allows you to identify forgotten syntax (Markdown mixup in RST). In RST, it's best to set roles explicitly.